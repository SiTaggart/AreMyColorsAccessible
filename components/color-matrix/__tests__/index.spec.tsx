/* eslint-env jest */

import React from 'react';
import ReactDOM from 'react-dom';
import { mount, ReactWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import ColorCombos from 'color-combos';
import ColorMatrix, { ColorMatrixProps } from '..';

describe('ColorMatrix', (): void => {
  const onColorChangeMock: jest.Mock = jest.fn();
  const mockColorColorCombos = ColorCombos(['#fff', '#ccc', '#777', '#000']);
  let mockProps: ColorMatrixProps;
  let wrapper: ReactWrapper;

  if (mockColorColorCombos !== false) {
    mockProps = {
      colors: ['#fff', '#ccc', '#777', '#000'],
      colorCombos: mockColorColorCombos,
      onColorChange: onColorChangeMock,
    };
  }

  beforeEach((): void => {
    jest.clearAllMocks();
    wrapper = mount(<ColorMatrix {...mockProps} onColorChange={onColorChangeMock} />);
  });

  it('renders without crashing', (): void => {
    ReactDOM.render(<ColorMatrix {...mockProps} />, document.createElement('div'));
    const colorCard = renderer.create(<ColorMatrix {...mockProps} />).toJSON();
    expect(colorCard).toMatchSnapshot();
  });

  it('should call the onColorChange method when a color is changed in an input', (): void => {
    wrapper
      .find('#colorhex-0')
      .at(0)
      .simulate('change', { target: { value: '#ccc' } });
    expect(onColorChangeMock).toHaveBeenCalledWith('#ccc', 0);
  });

  it.skip('should call the onColorChange method when a color is changed in on a slider', (): void => {
    wrapper
      .find('#hsl-0-Lightness')
      .at(0)
      .simulate('change', { target: { value: '12' } });
    expect(onColorChangeMock).toHaveBeenCalledWith('#1F1F1F', 0);
  });
});
