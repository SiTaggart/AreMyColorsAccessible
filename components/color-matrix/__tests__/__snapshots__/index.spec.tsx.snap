// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorMatrix renders without crashing 1`] = `
.emotion-386 {
  margin-top: 3rem;
  overflow-x: auto;
}

.emotion-384 {
  border-collapse: collapse;
  width: 100%;
}

.emotion-34 {
  font-weight: bold;
  padding: 0.5rem;
}

.emotion-34:first-of-type {
  padding-left: 0;
}

.emotion-0 {
  background: rgba(255,255,255,0.1);
  border: solid 1px;
  border-radius: 3px;
  font-size: 1.2rem;
  padding: 0.5rem 1rem;
  margin-bottom: 2rem;
  width: 100%;
  margin-bottom: 0.5rem;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  width: 100%;
}

@media (min-width:1024px) {
  .emotion-10 {
    -webkit-flex: 1 0 auto;
    -ms-flex: 1 0 auto;
    flex: 1 0 auto;
  }
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
}

.emotion-2 {
  display: inline-block;
  font-size: 1rem;
  margin-bottom: 0;
  width: 100%;
}

.emotion-8 {
  -webkit-flex: 6;
  -ms-flex: 6;
  flex: 6;
}

.emotion-6 {
  -webkit-appearance: none;
  background: transparent;
  border-radius: 2px;
  color: currentColor;
  display: block;
  margin: 10px 0;
  width: 100%;
}

.emotion-6::-webkit-slider-runnable-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-6::-webkit-slider-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
  -webkit-appearance: none;
  margin-top: -9px;
}

.emotion-6:focus::-webkit-slider-runnable-track {
  background: currentColor;
}

.emotion-6::-moz-range-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-6::-moz-range-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-6::-ms-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  background: transparent;
  border-color: transparent;
  border-width: 8px 0;
  color: transparent;
}

.emotion-6::-ms-fill-lower {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-6::-ms-fill-upper {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-6::-ms-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-6:focus::-ms-fill-lower {
  background: currentColor;
}

.emotion-6:focus::-ms-fill-upper {
  background: currentColor;
}

.emotion-148 {
  border: solid 1px #fff;
  padding: 0.5rem;
}

.emotion-164 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 8rem;
  min-width: 12rem;
  overflow: hidden;
  text-align: center;
  -webkit-transition: opacity ease-in;
  transition: opacity ease-in;
}

.emotion-152 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-150 {
  background-color: #F7F8F8;
  color: #666D70;
  border-radius: 3px;
  display: inline-block;
  font-size: 12px;
  padding: 1px 4px;
}

.emotion-154 {
  color: #FFFFFF;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-weight: bold;
  font-size: 2rem;
  text-align: center;
}

.emotion-158 {
  background-color: #FBDBDB;
  color: #C52020;
  border-radius: 3px;
  display: inline-block;
  font-size: 12px;
  padding: 1px 4px;
}

.emotion-178 {
  background-color: #CFFCDA;
  color: #004215;
  border-radius: 3px;
  display: inline-block;
  font-size: 12px;
  padding: 1px 4px;
}

.emotion-212 {
  color: #CCCCCC;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-weight: bold;
  font-size: 2rem;
  text-align: center;
}

.emotion-272 {
  color: #777777;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-weight: bold;
  font-size: 2rem;
  text-align: center;
}

.emotion-332 {
  color: #000000;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-weight: bold;
  font-size: 2rem;
  text-align: center;
}

<section
  className="emotion-386 emotion-387"
>
  <table
    className="emotion-384 emotion-385"
  >
    <thead>
      <tr
        className="emotion-144 emotion-145"
        data-testid="colorMatrix-tr"
      >
        <td
          style={
            Object {
              "width": "6rem",
            }
          }
        />
        <th
          className="emotion-34 emotion-35"
          data-testid="colorMatrix-th"
          scope="col"
        >
          <input
            aria-label="hex colour code"
            autoComplete="off"
            className="emotion-0 emotion-1"
            data-testid="colorhex-0"
            id="colorhex-0"
            onChange={[Function]}
            type="text"
            value="#fff"
          />
          <div
            className="emotion-32 emotion-33"
            data-testid="form-hsl-sliders"
          >
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-0-Hue"
                >
                  H
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-0-Hue"
                  id="hsl-0-Hue"
                  max={360}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-0-Saturation"
                >
                  S
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-0-Saturation"
                  id="hsl-0-Saturation"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-0-Lightness"
                >
                  L
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-0-Lightness"
                  id="hsl-0-Lightness"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={100}
                />
              </div>
            </div>
          </div>
        </th>
        <th
          className="emotion-34 emotion-35"
          data-testid="colorMatrix-th"
          scope="col"
        >
          <input
            aria-label="hex colour code"
            autoComplete="off"
            className="emotion-0 emotion-1"
            data-testid="colorhex-1"
            id="colorhex-1"
            onChange={[Function]}
            type="text"
            value="#ccc"
          />
          <div
            className="emotion-32 emotion-33"
            data-testid="form-hsl-sliders"
          >
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-1-Hue"
                >
                  H
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-1-Hue"
                  id="hsl-1-Hue"
                  max={360}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-1-Saturation"
                >
                  S
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-1-Saturation"
                  id="hsl-1-Saturation"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-1-Lightness"
                >
                  L
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-1-Lightness"
                  id="hsl-1-Lightness"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={80}
                />
              </div>
            </div>
          </div>
        </th>
        <th
          className="emotion-34 emotion-35"
          data-testid="colorMatrix-th"
          scope="col"
        >
          <input
            aria-label="hex colour code"
            autoComplete="off"
            className="emotion-0 emotion-1"
            data-testid="colorhex-2"
            id="colorhex-2"
            onChange={[Function]}
            type="text"
            value="#777"
          />
          <div
            className="emotion-32 emotion-33"
            data-testid="form-hsl-sliders"
          >
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-2-Hue"
                >
                  H
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-2-Hue"
                  id="hsl-2-Hue"
                  max={360}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-2-Saturation"
                >
                  S
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-2-Saturation"
                  id="hsl-2-Saturation"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-2-Lightness"
                >
                  L
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-2-Lightness"
                  id="hsl-2-Lightness"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={47}
                />
              </div>
            </div>
          </div>
        </th>
        <th
          className="emotion-34 emotion-35"
          data-testid="colorMatrix-th"
          scope="col"
        >
          <input
            aria-label="hex colour code"
            autoComplete="off"
            className="emotion-0 emotion-1"
            data-testid="colorhex-3"
            id="colorhex-3"
            onChange={[Function]}
            type="text"
            value="#000"
          />
          <div
            className="emotion-32 emotion-33"
            data-testid="form-hsl-sliders"
          >
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-3-Hue"
                >
                  H
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-3-Hue"
                  id="hsl-3-Hue"
                  max={360}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-3-Saturation"
                >
                  S
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-3-Saturation"
                  id="hsl-3-Saturation"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
            <div
              className="emotion-10 emotion-11"
            >
              <div
                className="emotion-4 emotion-5"
              >
                <label
                  className="emotion-2 emotion-3"
                  htmlFor="hsl-3-Lightness"
                >
                  L
                </label>
              </div>
              <div
                className="emotion-8 emotion-9"
              >
                <input
                  className="emotion-6 emotion-7"
                  data-testid="hsl-3-Lightness"
                  id="hsl-3-Lightness"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onInput={[Function]}
                  type="range"
                  value={0}
                />
              </div>
            </div>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="emotion-144 emotion-145"
        data-testid="colorMatrix-tr"
      >
        <th
          className="emotion-34 emotion-35"
          data-testid="colorMatrix-th"
          scope="row"
        >
          #FFFFFF
        </th>
        <td
          className="emotion-148 emotion-149"
        >
           
        </td>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#CCCCCC",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                1.61
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-154 emotion-155"
                color="#FFFFFF"
                data-testid="colorCard-swatch"
              >
                Nope
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-158 emotion-151"
              >
                Small: 
                Fail
              </span>
              <span
                className="emotion-158 emotion-151"
              >
                Large: 
                Fail
              </span>
            </div>
          </div>
        </td>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#777777",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                4.48
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-154 emotion-155"
                color="#FFFFFF"
                data-testid="colorCard-swatch"
              >
                Kinda
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-158 emotion-151"
              >
                Small: 
                Fail
              </span>
              <span
                className="emotion-178 emotion-151"
              >
                Large: 
                AA
              </span>
            </div>
          </div>
        </td>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#000000",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                21
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-154 emotion-155"
                color="#FFFFFF"
                data-testid="colorCard-swatch"
              >
                Yup
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-178 emotion-151"
              >
                Small: 
                AAA
              </span>
              <span
                className="emotion-178 emotion-151"
              >
                Large: 
                AAA
              </span>
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="emotion-144 emotion-145"
        data-testid="colorMatrix-tr"
      >
        <th
          className="emotion-34 emotion-35"
          data-testid="colorMatrix-th"
          scope="row"
        >
          #CCCCCC
        </th>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#FFFFFF",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                1.61
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-212 emotion-155"
                color="#CCCCCC"
                data-testid="colorCard-swatch"
              >
                Nope
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-158 emotion-151"
              >
                Small: 
                Fail
              </span>
              <span
                className="emotion-158 emotion-151"
              >
                Large: 
                Fail
              </span>
            </div>
          </div>
        </td>
        <td
          className="emotion-148 emotion-149"
        >
           
        </td>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#777777",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                2.79
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-212 emotion-155"
                color="#CCCCCC"
                data-testid="colorCard-swatch"
              >
                Nope
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-158 emotion-151"
              >
                Small: 
                Fail
              </span>
              <span
                className="emotion-158 emotion-151"
              >
                Large: 
                Fail
              </span>
            </div>
          </div>
        </td>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#000000",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                13.08
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-212 emotion-155"
                color="#CCCCCC"
                data-testid="colorCard-swatch"
              >
                Yup
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-178 emotion-151"
              >
                Small: 
                AAA
              </span>
              <span
                className="emotion-178 emotion-151"
              >
                Large: 
                AAA
              </span>
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="emotion-144 emotion-145"
        data-testid="colorMatrix-tr"
      >
        <th
          className="emotion-34 emotion-35"
          data-testid="colorMatrix-th"
          scope="row"
        >
          #777777
        </th>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#FFFFFF",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                4.48
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-272 emotion-155"
                color="#777777"
                data-testid="colorCard-swatch"
              >
                Kinda
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-158 emotion-151"
              >
                Small: 
                Fail
              </span>
              <span
                className="emotion-178 emotion-151"
              >
                Large: 
                AA
              </span>
            </div>
          </div>
        </td>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#CCCCCC",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                2.79
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-272 emotion-155"
                color="#777777"
                data-testid="colorCard-swatch"
              >
                Nope
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-158 emotion-151"
              >
                Small: 
                Fail
              </span>
              <span
                className="emotion-158 emotion-151"
              >
                Large: 
                Fail
              </span>
            </div>
          </div>
        </td>
        <td
          className="emotion-148 emotion-149"
        >
           
        </td>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#000000",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                4.69
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-272 emotion-155"
                color="#777777"
                data-testid="colorCard-swatch"
              >
                Yup
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-178 emotion-151"
              >
                Small: 
                AA
              </span>
              <span
                className="emotion-178 emotion-151"
              >
                Large: 
                AAA
              </span>
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="emotion-144 emotion-145"
        data-testid="colorMatrix-tr"
      >
        <th
          className="emotion-34 emotion-35"
          data-testid="colorMatrix-th"
          scope="row"
        >
          #000000
        </th>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#FFFFFF",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                21
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-332 emotion-155"
                color="#000000"
                data-testid="colorCard-swatch"
              >
                Yup
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-178 emotion-151"
              >
                Small: 
                AAA
              </span>
              <span
                className="emotion-178 emotion-151"
              >
                Large: 
                AAA
              </span>
            </div>
          </div>
        </td>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#CCCCCC",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                13.08
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-332 emotion-155"
                color="#000000"
                data-testid="colorCard-swatch"
              >
                Yup
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-178 emotion-151"
              >
                Small: 
                AAA
              </span>
              <span
                className="emotion-178 emotion-151"
              >
                Large: 
                AAA
              </span>
            </div>
          </div>
        </td>
        <td
          className="emotion-148 emotion-149"
          style={
            Object {
              "backgroundColor": "#777777",
            }
          }
        >
          <div
            className="emotion-164 emotion-165"
            data-testid="colorCard"
          >
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-150 emotion-151"
                title="Color contrast ratio"
              >
                4.69
                 : 1
              </span>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <div
                className="emotion-332 emotion-155"
                color="#000000"
                data-testid="colorCard-swatch"
              >
                Yup
              </div>
            </div>
            <div
              className="emotion-152 emotion-153"
            >
              <span
                className="emotion-178 emotion-151"
              >
                Small: 
                AA
              </span>
              <span
                className="emotion-178 emotion-151"
              >
                Large: 
                AAA
              </span>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
